### PR TITLE
feature/504 459

### PR DESCRIPTION
**Pull request by gentee**
Added changes from 504 and 459 tasks

### CODE REVIEW

**Review by potehinre — APPROVED** [review]

File: packages/parser/common_parse_data_full.go [hunk 1]
```diff
@@ -625,12 +625,10 @@ func (b *Block) playBlock(dbTransaction *model.DbTransaction) error {
 		)
 		p.DbTransaction = dbTransaction
 
-		if b.GenBlock {
-			err = dbTransaction.Connection().Exec(fmt.Sprintf("SAVEPOINT \"tx-%d\";", curTx)).Error
-			if err != nil {
-				logger.WithFields(log.Fields{"type": consts.DBError, "error": err, "tx_hash": p.TxHash}).Error("using savepoint")
-				return err
-			}
+		err = dbTransaction.Connection().Exec(fmt.Sprintf("SAVEPOINT \"tx-%d\";", curTx)).Error
```
Comment (ghost) [inline packages/parser/common_parse_data_full.go]: why are you use sql queries outside from models package?

File: packages/parser/common_parse_data_full.go [hunk 2]
```diff
@@ -640,11 +638,11 @@ func (b *Block) playBlock(dbTransaction *model.DbTransaction) error {
...
 				b.StopCount = curTx
 				model.IncrementTxAttemptCount(p.TxHash)
-				err = dbTransaction.Connection().Exec(fmt.Sprintf("ROLLBACK TO SAVEPOINT \"tx-%d\";", curTx)).Error
-				if err != nil {
-					logger.WithFields(log.Fields{"type": consts.DBError, "error": err, "tx_hash": p.TxHash}).Error("rolling back to previous savepoint")
-				}
-				return err
+			}
+			errRoll := dbTransaction.Connection().Exec(fmt.Sprintf("ROLLBACK TO SAVEPOINT \"tx-%d\";", curTx)).Error
+			if errRoll != nil {
```
Comment (ghost) [inline packages/parser/common_parse_data_full.go]: one more sql outside models package

File: packages/parser/common_parse_data_full.go [hunk 3]
```diff
@@ -657,6 +655,10 @@ func (b *Block) playBlock(dbTransaction *model.DbTransaction) error {
 			}
 			continue
 		}
+		err = dbTransaction.Connection().Exec(fmt.Sprintf("RELEASE SAVEPOINT \"tx-%d\";", curTx)).Error
```
Comment (ghost) [inline packages/parser/common_parse_data_full.go]: the same stuff

**Review by ghost — CHANGES_REQUESTED** [review]

File: packages/parser/common_parse_data_full.go (hunk 1 above)
Comment (gentee) [reply packages/parser/common_parse_data_full.go]: Moved savepoint functions to db model

**Review by ghost — APPROVED** [review]

**Pull request merged**